src/web/error.rs

```diff
@@ -132,6 +136,110 @@ impl From<PoolError> for IronError {
+                AxumErrorPage {
+                    title: "Requested resource does not exist",
+                    message: Some("no such resource".into()),
+                    status: StatusCode::NOT_FOUND,
+                }
+                .into_response()
+            }
+
+            AxumNope::BuildNotFound => AxumErrorPage {
+                title: "The requested build does not exist",
+                message: Some("no such build".into()),
+                status: StatusCode::NOT_FOUND,
+            }
+            .into_response(),
+
+            AxumNope::CrateNotFound => {
+                // user tried to navigate to a crate that doesn't exist
+                // TODO: Display the attempted crate and a link to a search for said crate
+                AxumErrorPage {
+                    title: "The requested crate does not exist",
+                    message: Some("no such crate".into()),
+                    status: StatusCode::NOT_FOUND,
+                }
+                .into_response()
+            }
+
+            AxumNope::OwnerNotFound => AxumErrorPage {
+                title: "The requested owner does not exist",
+                message: Some("no such owner".into()),
+                status: StatusCode::NOT_FOUND,
+            }
+            .into_response(),
+
+            AxumNope::VersionNotFound => {
+                // user tried to navigate to a crate with a version that does not exist
+                // TODO: Display the attempted crate and version
+                AxumErrorPage {
+                    title: "The requested version does not exist",
+                    message: Some("no such version for this crate".into()),
+                    status: StatusCode::NOT_FOUND,
+                }
+                .into_response()
+            }
+            AxumNope::NoResults => {
+                todo!("to be implemented when search-handler is migrated to axum")
+            }
+            AxumNope::InternalServerError => {
+                // something went wrong, details should have been logged
+                AxumErrorPage {
+                    title: "Internal server error",
+                    message: Some("internal server error".into()),
+                    status: StatusCode::INTERNAL_SERVER_ERROR,
+                }
+                .into_response()
+            }
+            AxumNope::InternalError(source) => generate_internal_error_page(source).into_response(),
+            AxumNope::InternalTokio(source) => generate_internal_error_page(source).into_response(),
+        }
+    }
+}
+
+fn generate_internal_error_page<E: Into<anyhow::Error>>(error: E) -> impl IntoResponse {
+    let error = anyhow::anyhow!(error);
+
+    let web_error = crate::web::AxumErrorPage {
+        title: "Internal Server Error",
+        message: ::std::option::Option::Some(::std::borrow::Cow::Owned(error.to_string())),
+        status: ::http::StatusCode::INTERNAL_SERVER_ERROR,
```

Comment:
```suggestion
        message: Some(Cow::Owned(error.to_string())),
        status: StatusCode::INTERNAL_SERVER_ERROR,
```

Reply:
valid, I probably copied this from a macro